### PR TITLE
feat: Show connected destinations in actions ui

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Want to migrate people away from action webhooks so we can make it simple by showing the new apps under actions

## Changes

* Fixes some bad logic that wasn't checking the "free" status each time
* Adds a new "connected destinations" section to actions if hog functions are enabled
* Hides action webhooks if hog functions are enabled unless it is active in which case shows a deprecation notice.
* Hides the integration area for webhooks if hog functions enabled

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/scenes/actions/actionLogic.ts [hunk 1]
```diff
@@ -35,11 +35,11 @@ export const actionLogic = kea<actionLogicType>([
...
         ],
-        matchingPluginConfigs: [
-            null as PluginConfigWithPluginInfoNew[] | null,
+        matchingHogFunctions: [
+            null as HogFunctionType[] | null,
             {
-                loadMatchingPluginConfigs: async () => {
-                    const res = await api.actions.listMatchingPluginConfigs(props.id!)
+                loadMatchingHogFunctions: async () => {
+                    const res = await api.hogFunctions.list({ filters: { actions: [{ id: `${props.id}` }] } })
```
Comment (mariusandra) [inline frontend/src/scenes/actions/actionLogic.ts:42]: There's some magic making this work. Will it only work if the action we're interested in is the first one, or will it match all?

File: posthog/api/hog_function.py [hunk 2]
```diff
@@ -213,6 +214,13 @@ def safely_get_queryset(self, queryset: QuerySet) -> QuerySet:
         if self.action == "list":
             queryset = queryset.filter(deleted=False)
 
+        if self.request.GET.get("filters"):
+            try:
+                filters = json.loads(self.request.GET["filters"])
+                queryset = queryset.filter(filters__contains=filters)
+            except Exception:
+                raise exceptions.ValidationError({"filter": f"Invalid filter"})
```
Comment (mariusandra) [inline posthog/api/hog_function.py:222]: No tests? 🙈

**Review by mariusandra — APPROVED** [review]
Some questions and a ✅ to move things along.

File: posthog/api/hog_function.py (hunk 2 above)
Comment (benjackwhite) [reply posthog/api/hog_function.py:222]: NO YOU'RE NO TESTS 😉

File: frontend/src/scenes/actions/actionLogic.ts (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/scenes/actions/actionLogic.ts:42]: Added a test - it actually does work!

**Pull request merged**